Forwarded: https://github.com/BuddiesOfBudgie/budgie-desktop/pull/124
Author: David Mohammed <fossfreedom@ubuntu.com>
Last-Update: 2022-04-06
Description: [PATCH1/5] Regenerate vapi for mutter 10 This commit updates the vapi
 for mutter 10/clutter et al since the stable release of GNOME 42. This also
 fixes an existing issue with our generation of CursorTracker.get_pointer
 where params both return values.

---
 vapi/Meta-10.metadata       |  1 +
 vapi/libmutter-10.vapi      |  3 ++-
 vapi/mutter-clutter-10.vapi | 37 +++++++++++++++----------------------
 vapi/mutter-cogl-10.vapi    |  2 +-
 4 files changed, 19 insertions(+), 24 deletions(-)

diff --git a/vapi/Meta-10.metadata b/vapi/Meta-10.metadata
index c3a6c902..50f8884d 100644
--- a/vapi/Meta-10.metadata
+++ b/vapi/Meta-10.metadata
@@ -11,6 +11,7 @@ CloseDialog cheader_filename="meta/meta-close-dialog.h"
 CursorTracker cheader_filename="meta/meta-cursor-tracker.h"
 CursorTracker.get_for_display parent="Meta.Display" symbol_type="method" instance_idx=0 name="get_cursor_tracker"
 CursorTracker.get_pointer.mods out
+CursorTracker.get_pointer.coords out
 Display cheader_filename="meta/display.h"
 Dnd cheader_filename="meta/meta-dnd.h"
 IdleMonitor cheader_filename="meta/meta-idle-monitor.h"
diff --git a/vapi/libmutter-10.vapi b/vapi/libmutter-10.vapi
index 46b71c4d..16dc91e7 100644
--- a/vapi/libmutter-10.vapi
+++ b/vapi/libmutter-10.vapi
@@ -314,7 +314,7 @@ namespace Meta {
 		[CCode (has_construct_function = false)]
 		protected CursorTracker ();
 		public void get_hot (out int x, out int y);
-		public void get_pointer (Graphene.Point coords, out Clutter.ModifierType mods);
+		public void get_pointer (out Graphene.Point coords, out Clutter.ModifierType mods);
 		public bool get_pointer_visible ();
 		public float get_scale ();
 		public unowned Cogl.Texture get_sprite ();
@@ -823,6 +823,7 @@ namespace Meta {
 		public uint user_time { get; }
 		public Meta.WindowType window_type { get; }
 		public string wm_class { get; }
+		public signal void monitor_changed (int old_monitor);
 		public signal void position_changed ();
 		public signal void raised ();
 		public signal void shown ();
diff --git a/vapi/mutter-clutter-10.vapi b/vapi/mutter-clutter-10.vapi
index 086468f9..cb4cb33c 100644
--- a/vapi/mutter-clutter-10.vapi
+++ b/vapi/mutter-clutter-10.vapi
@@ -5301,7 +5301,6 @@ namespace Clutter {
 	[Version (since = "0.2")]
 	public class AnyEvent : Clutter.Event {
 		public Clutter.EventFlags flags;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -5444,7 +5443,6 @@ namespace Clutter {
 		public uint32 evdev_code;
 		public Clutter.EventFlags flags;
 		public Clutter.ModifierType modifier_state;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -5558,7 +5556,6 @@ namespace Clutter {
 		public Clutter.EventFlags flags;
 		public weak Clutter.Actor related;
 		public weak Clutter.EventSequence sequence;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -5596,7 +5593,6 @@ namespace Clutter {
 	public class DeviceEvent : Clutter.Event {
 		public weak Clutter.InputDevice device;
 		public Clutter.EventFlags flags;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -5794,7 +5790,9 @@ namespace Clutter {
 	[CCode (cheader_filename = "clutter/clutter.h", has_type_id = false)]
 	[Compact]
 	public class Frame {
+		public Clutter.FrameHint get_hints ();
 		public bool has_result ();
+		public void set_hint (Clutter.FrameHint hint);
 		public void set_result (Clutter.FrameResult result);
 	}
 	[CCode (cheader_filename = "clutter/clutter.h", type_id = "clutter_frame_clock_get_type ()")]
@@ -5806,7 +5804,7 @@ namespace Clutter {
 		public float get_refresh_rate ();
 		public void inhibit ();
 		public void notify_ready ();
-		public void record_flip_time (int64 flip_time_us);
+		public void record_flip (int64 flip_time_us, Clutter.FrameHint hints);
 		public void remove_timeline (Clutter.Timeline timeline);
 		public void schedule_update ();
 		public void schedule_update_now ();
@@ -5916,7 +5914,6 @@ namespace Clutter {
 		public uint32 len;
 		public Clutter.PreeditResetMode mode;
 		public int32 offset;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public weak string text;
 		public uint32 time;
@@ -6107,7 +6104,6 @@ namespace Clutter {
 		public uint16 hardware_keycode;
 		public uint keyval;
 		public Clutter.ModifierType modifier_state;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -6189,7 +6185,6 @@ namespace Clutter {
 		public double dy_unaccel;
 		public Clutter.EventFlags flags;
 		public Clutter.ModifierType modifier_state;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public int64 time_us;
@@ -6218,7 +6213,6 @@ namespace Clutter {
 		public Clutter.EventFlags flags;
 		public uint32 group;
 		public uint32 mode;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -6233,7 +6227,6 @@ namespace Clutter {
 		public uint32 mode;
 		public uint32 ring_number;
 		public Clutter.InputDevicePadSource ring_source;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -6245,7 +6238,6 @@ namespace Clutter {
 		public Clutter.EventFlags flags;
 		public uint32 group;
 		public uint32 mode;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 strip_number;
 		public Clutter.InputDevicePadSource strip_source;
@@ -6412,7 +6404,6 @@ namespace Clutter {
 	public class ProximityEvent : Clutter.Event {
 		public weak Clutter.InputDevice device;
 		public Clutter.EventFlags flags;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -6484,7 +6475,6 @@ namespace Clutter {
 		public Clutter.EventFlags flags;
 		public Clutter.ModifierType modifier_state;
 		public Clutter.ScrollSource scroll_source;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -6628,6 +6618,7 @@ namespace Clutter {
 		public unowned Clutter.Actor get_actor_at_pos (Clutter.PickMode pick_mode, float x, float y);
 		public bool get_capture_final_size (Cairo.RectangleInt rect, out int out_width, out int out_height, out float out_scale);
 		public unowned Clutter.Actor get_device_actor (Clutter.InputDevice device, Clutter.EventSequence? sequence);
+		public unowned Clutter.Actor get_event_actor (Clutter.Event event);
 		public int64 get_frame_counter ();
 		public unowned Clutter.Actor get_grab_actor ();
 		[Version (since = "0.6")]
@@ -6636,7 +6627,7 @@ namespace Clutter {
 		public Clutter.Perspective get_perspective ();
 		[Version (since = "0.4")]
 		public unowned string get_title ();
-		public unowned Clutter.Grab grab (Clutter.Actor actor);
+		public Clutter.Grab grab (Clutter.Actor actor);
 		public bool paint_to_buffer (Cairo.RectangleInt rect, float scale, [CCode (array_length = false)] uint8[] data, int stride, Cogl.PixelFormat format, Clutter.PaintFlag paint_flags) throws GLib.Error;
 		public Clutter.Content paint_to_content (Cairo.RectangleInt rect, float scale, Clutter.PaintFlag paint_flags) throws GLib.Error;
 		public void paint_to_framebuffer (Cogl.Framebuffer framebuffer, Cairo.RectangleInt rect, float scale, Clutter.PaintFlag paint_flags);
@@ -7036,7 +7027,6 @@ namespace Clutter {
 		public Clutter.EventFlags flags;
 		public Clutter.ModifierType modifier_state;
 		public weak Clutter.EventSequence sequence;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -7056,7 +7046,6 @@ namespace Clutter {
 		public uint n_fingers;
 		public Clutter.TouchpadGesturePhase phase;
 		public float scale;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -7074,7 +7063,6 @@ namespace Clutter {
 		public Clutter.EventFlags flags;
 		public uint n_fingers;
 		public Clutter.TouchpadGesturePhase phase;
-		public weak Clutter.Actor source;
 		public weak Clutter.Stage stage;
 		public uint32 time;
 		public Clutter.EventType type;
@@ -7407,7 +7395,6 @@ namespace Clutter {
 		public uint32 time;
 		public Clutter.EventFlags flags;
 		public weak Clutter.Stage stage;
-		public weak Clutter.Actor source;
 		public Clutter.TouchpadGesturePhase phase;
 		public uint32 n_fingers;
 		public float x;
@@ -7677,6 +7664,12 @@ namespace Clutter {
 		HORIZONTAL,
 		VERTICAL
 	}
+	[CCode (cheader_filename = "clutter/clutter.h", cprefix = "CLUTTER_FRAME_HINT_", type_id = "clutter_frame_hint_get_type ()")]
+	[Flags]
+	public enum FrameHint {
+		NONE,
+		DIRECT_SCANOUT_ATTEMPTED
+	}
 	[CCode (cheader_filename = "clutter/clutter.h", cprefix = "CLUTTER_FRAME_INFO_FLAG_", type_id = "clutter_frame_info_flag_get_type ()")]
 	[Flags]
 	public enum FrameInfoFlag {
@@ -8185,13 +8178,13 @@ namespace Clutter {
 		POINTER,
 		TOUCHSCREEN
 	}
-	[CCode (cheader_filename = "clutter/clutter.h", cprefix = "CLUTTER_IMAGE_ERROR_INVALID_")]
+	[CCode (cheader_filename = "clutter/clutter.h", cprefix = "CLUTTER_IMAGE_ERROR_INVALID_", type_id = "clutter_image_error_get_type ()")]
 	[Version (since = "1.10")]
 	public errordomain ImageError {
 		DATA;
 		public static GLib.Quark quark ();
 	}
-	[CCode (cheader_filename = "clutter/clutter.h", cprefix = "CLUTTER_SCRIPT_ERROR_INVALID_")]
+	[CCode (cheader_filename = "clutter/clutter.h", cprefix = "CLUTTER_SCRIPT_ERROR_INVALID_", type_id = "clutter_script_error_get_type ()")]
 	[Version (since = "0.6")]
 	public errordomain ScriptError {
 		TYPE_FUNCTION,
@@ -8207,9 +8200,9 @@ namespace Clutter {
 	public delegate bool BindingActionFunc (GLib.Object gobject, string action_name, uint key_val, Clutter.ModifierType modifiers);
 	[CCode (cheader_filename = "clutter/clutter.h", instance_pos = 1.9)]
 	public delegate void Callback (Clutter.Actor actor);
-	[CCode (cheader_filename = "clutter/clutter.h", instance_pos = 1.9)]
+	[CCode (cheader_filename = "clutter/clutter.h", instance_pos = 2.9)]
 	[Version (since = "1.18")]
-	public delegate bool EventFilterFunc (Clutter.Event event);
+	public delegate bool EventFilterFunc (Clutter.Event event, Clutter.Actor event_actor);
 	[CCode (cheader_filename = "clutter/clutter.h", instance_pos = 1.9)]
 	[Version (since = "1.0")]
 	public delegate void PathCallback (Clutter.PathNode node);
diff --git a/vapi/mutter-cogl-10.vapi b/vapi/mutter-cogl-10.vapi
index ac7a6525..ad3c2d6f 100644
--- a/vapi/mutter-cogl-10.vapi
+++ b/vapi/mutter-cogl-10.vapi
@@ -884,7 +884,7 @@ namespace Cogl {
 		SYNC_AND_COMPLETE_EVENT,
 		N_FEATURES
 	}
-	[CCode (cheader_filename = "cogl/cogl.h", cprefix = "COGL_SCANOUT_ERROR_")]
+	[CCode (cheader_filename = "cogl/cogl.h", cprefix = "COGL_SCANOUT_ERROR_", has_type_id = false)]
 	public errordomain ScanoutError {
 		[CCode (cname = "COGL_SCANOUT_ERROR_INHIBITED")]
 		SCANOUT_ERROR_INHIBITED;
-- 
2.34.1

